### PR TITLE
Create a proper infinite grid helper

### PR DESCRIPTION
**Pull request by AndrewPrifer**

### CODE REVIEW

**Comment by gsimone** [comment]
Hello, I'm mr. steal-your-code-and-add-it-to-drei

File: packages/r3f/src/InfiniteGridHelper/index.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        uSubgridOpacity: {
+          value: subgridOpacity,
+        },
+        uGridOpacity: {
+          value: gridOpacity,
+        },
+      },
+      transparent: true,
+      // language=GLSL
+      vertexShader: `
```
Comment (gsimone) [inline packages/r3f/src/InfiniteGridHelper/index.ts:47]:
```suggestion
      vertexShader: /* glsl */`
```

File: packages/r3f/src/InfiniteGridHelper/index.ts [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+          void main() {
+              // Scale the plane by the drawing distance
+              worldPosition = position.xzy * uDistance;
+              worldPosition.xz += cameraPosition.xz;
+
+              gl_Position = projectionMatrix * modelViewMatrix * vec4(worldPosition, 1.0);
+          }
+      `,
+      // language=GLSL
+      fragmentShader: `
```
Comment (gsimone) [inline packages/r3f/src/InfiniteGridHelper/index.ts:60]:
```suggestion
      fragmentShader: /* glsl */`
```

**Comment by gsimone** [comment]
Actually this might be a good thing to add to three.js itself!

File: packages/r3f/src/components/SnapshotEditor.tsx [hunk 3]
```diff
@@ -50,9 +51,11 @@ const EditorScene: React.FC<{snapshotEditorSheet: ISheet; paneId: string}> = ({
   const showGrid = useVal(editorObject?.props.viewport.showGrid) ?? true
   const showAxes = useVal(editorObject?.props.viewport.showAxes) ?? true
 
+  const grid = useMemo(() => new InfiniteGridHelper(), [])
+
   return (
     <>
-      {showGrid && <gridHelper args={[20, 20, '#6e6e6e', '#4a4b4b']} />}
+      {showGrid && <primitive object={grid} />}
```
Comment (gsimone) [inline packages/r3f/src/components/SnapshotEditor.tsx:58]: I think you could use extend here

```
extend({ InfiniteGridHelper })

...
<infiniteGridHelper />
```

Comment (AndrewPrifer) [reply packages/r3f/src/components/SnapshotEditor.tsx:58]: Oh yeah it was complaining it was not part of the THREE namespace and I was too lazy to do anything about it

**Review by AriaMinaei — APPROVED** [review]
Awesome work! Ready to merge.

**Pull request merged**